File: Partie-Backend/src/main/java/com/basketapp/basket_app/models/Comment.java
package com.basketapp.basket_app.models;

import java.util.Date;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.OneToOne;
import javax.persistence.PrePersist;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import com.basketapp.basket_app.dto.UtilisateurDto;
import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnore;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@NoArgsConstructor
@AllArgsConstructor
@Data
public class Comment {
    @Id
    @GeneratedValue (strategy = GenerationType.IDENTITY)
    private  Long idComment;
    private String label;
    private Integer likes;
    private Integer dislikes;

    @Temporal(TemporalType.TIMESTAMP)
    @Column(nullable = false)
    @JsonFormat(pattern="yyyy-MM-dd")
    private Date commentdate;

    @PrePersist
     private void onCreate() {
        commentdate = new Date();
    }
    
    @JsonIgnore
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "actualite_id")
    private Actualite actualite;

    @OneToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "utilisateurDto_id")
    private UtilisateurDto userdto;
}

File: Partie-Mobile/lib/models/comment.dart
class comment {
  int? idComment;
  String? label;
  int? likes;
  int? dislikes;
  String? commentdate;
  Userdto? userdto;

  comment(
      {this.idComment,
      this.label,
      this.likes,
      this.dislikes,
      this.commentdate,
      this.userdto,
      });

  comment.fromJson(Map<String, dynamic> json) {
    idComment = json['idComment'];
    label = json['label'];
    likes = json['likes'];
    dislikes = json['dislikes'];
    commentdate = json['commentdate'];
    userdto =
        json['userdto'] != null ? new Userdto.fromJson(json['userdto']) : null;
    
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['idComment'] = this.idComment;
    data['label'] = this.label;
    data['likes'] = this.likes;
    data['dislikes'] = this.dislikes;
    data['commentdate'] = this.commentdate;
    if (this.userdto != null) {
      data['userdto'] = this.userdto!.toJson();
    }
    return data;
  }
}

class Userdto {
  int? idUtil;
  String? nomUtil;
  String? prenomUtil;

  Userdto({this.idUtil, this.nomUtil, this.prenomUtil});

  Userdto.fromJson(Map<String, dynamic> json) {
    idUtil = json['idUtil'];
    nomUtil = json['nomUtil'];
    prenomUtil = json['prenomUtil'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['idUtil'] = this.idUtil;
    data['nomUtil'] = this.nomUtil;
    data['prenomUtil'] = this.prenomUtil;
    return data;
  }
}
